File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/list/wordsearch/WordSearchResultYearMonthListItem.java
package com.websarva.wings.android.zuboradiary.ui.list.wordsearch;

import com.websarva.wings.android.zuboradiary.ui.list.DiaryYearMonthListAdapter.ViewType;
import com.websarva.wings.android.zuboradiary.ui.list.DiaryYearMonthListItemBase;

import java.time.YearMonth;
import java.util.Objects;

public class WordSearchResultYearMonthListItem extends DiaryYearMonthListItemBase {
    private final WordSearchResultDayList wordSearchResultDayList;

    WordSearchResultYearMonthListItem(ViewType viewType) {
        super(viewType);
        wordSearchResultDayList = new WordSearchResultDayList();
    }

    WordSearchResultYearMonthListItem(
            YearMonth yearMonth, WordSearchResultDayList wordSearchResultDayList) {
        super(yearMonth, ViewType.DIARY);

        Objects.requireNonNull(wordSearchResultDayList);
        if (wordSearchResultDayList.getWordSearchResultDayListItemList().isEmpty()) {
            throw new IllegalArgumentException();
        }

        this.wordSearchResultDayList = wordSearchResultDayList;
    }

    public WordSearchResultDayList getWordSearchResultDayList() {
        return wordSearchResultDayList;
    }
}

File: app/src/main/java/com/websarva/wings/android/zuboradiary/ui/list/wordsearch/WordSearchResultYearMonthListItem.kt
package com.websarva.wings.android.zuboradiary.ui.list.wordsearch

import com.websarva.wings.android.zuboradiary.ui.list.DiaryYearMonthListBaseAdapter
import com.websarva.wings.android.zuboradiary.ui.list.DiaryYearMonthListBaseItem
import java.time.YearMonth

internal class WordSearchResultYearMonthListItem : DiaryYearMonthListBaseItem {
    val wordSearchResultDayList: WordSearchResultDayList

    constructor(viewType: DiaryYearMonthListBaseAdapter.ViewType) : super(viewType) {
        this.wordSearchResultDayList = WordSearchResultDayList()
    }

    constructor(
        yearMonth: YearMonth,
        wordSearchResultDayList: WordSearchResultDayList
    ) : super(yearMonth, DiaryYearMonthListBaseAdapter.ViewType.DIARY) {
        require(wordSearchResultDayList.wordSearchResultDayListItemList.isNotEmpty())

        this.wordSearchResultDayList = wordSearchResultDayList
    }
}
